src/components/screen/__tests__/Chat.test.tsx

```diff
@@ -41,6 +44,48 @@ describe('[Chat] interaction', () => {
     fireEvent.press(chatBtn);
   });
 
+  it('should open image library when pressing photo icon button', () => {
+    const giftedChatprops = createTestProps();
+    const giftedChatComponent = createTestElement(
+      <GiftedChat {...giftedChatprops} />,
+    );
+    const giftedChatTestingLib = render(giftedChatComponent);
+    const menuBtn = giftedChatTestingLib.getByTestId('touch_menu');
+    const openImageLibFunction = jest.fn();
+    act(() => {
+      fireEvent.press(menuBtn);
+    });
+    waitForElement(() => {
```

Comment:
`waitForElement` is an async function and its parameter callback is run immediately instead of wait some element. 
You can change this code to below. 
```
const photoBtn = await waitForElement(() => testingLib.getByTestId('icon_photo'));
...
```
or
```
waitForElement(() => testingLib.getByTestId('icon_photo')).then( (photoBtn) => { 
... 
});
```
Refer to [this](https://www.native-testing-library.com/docs/api-async#waitforelement).

Reply:
actually I tried that. but got this error😯

```ts
  it('should open image library when pressing photo icon button', async () => {
    const giftedChatprops = createTestProps();
    const giftedChatComponent = createTestElement(
      <GiftedChat {...giftedChatprops} />,
    );
    const giftedChatTestingLib = render(giftedChatComponent);
    const menuBtn = giftedChatTestingLib.getByTestId('touch_menu');
    const openImageLibFunction = jest.fn();

    act(() => {
      fireEvent.press(menuBtn);
    });
    const photoBtn = await waitForElement(() =>
      testingLib.getByTestId('icon_photo'),
    );
    console.log('photoBtn!', photoBtn);
    act(() => {
      fireEvent.press(photoBtn);
      openImageLibFunction();
    });
    expect(openImageLibFunction).toHaveBeenCalled();
  });
```

`Unable to find an element with the testID of: icon_photo`

And it worked expected only when executed inside the callback function.

Reply:
@JongtaekChoi I understood what you said. thank you👍